.github/workflows/check-types.yml

```diff
@@ -0,0 +1,17 @@
+name: Check Types
```

Comment:
if we add a build status badge in the readme, this seems like we would have separate badges for Types, Linting and Unit tests?

Do we add only the unit tests badge in the readme then, or all of them?

Reply:
The trade-off is instantly knowing which step failed versus having to go and look at the logs. As far as `main` is concerned, Github keeps track as to whether or not _all_ of the steps passed or failed, so we can use that.